jobs/src/main/java/com/marklogic/spring/batch/config/LoadDocumentsFromDirectoryConfig.java

```
@@ -0,0 +1,97 @@
+import org.springframework.batch.item.file.ResourcesItemReader;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.Resource;
+import org.w3c.dom.Document;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.InputSource;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+
+/**
+ * Created by sstafford on 6/17/2016.
+ */
+public class LoadDocumentsFromDirectoryConfig extends AbstractMarkLogicBatchConfig {
+
+    @Bean
+    public Job job(@Qualifier("step1") Step step1) {
+        return jobBuilderFactory.get("loadImagesFromDirectoryJob").start(step1).build();
+    }
+
+    @Bean
+    @JobScope
+    protected Step step1(ItemReader<Resource> reader, ItemProcessor<Resource, Document> itemProcessor, ItemWriter<Document> itemWriter) {
+        return stepBuilderFactory.get("step1")
+                .<Resource, Document>chunk(getChunkSize())
+                .reader(reader)
+                .processor(itemProcessor)
+                .writer(itemWriter)
+                .build();
+    }
+
+    @Bean
+    @StepScope
+    public ItemReader<Resource> reader(
+            @Value("#{jobParameters['input_file_path']}") String inputFilePath)
+    {
+        logger.info("READER");
+        ResourcesItemReader itemReader = new ResourcesItemReader();
+        ArrayList<Resource> resourceList = new ArrayList<Resource>();
+        try {
+            Resource[] resources = ctx.getResources(inputFilePath);
+            for (int i = 0; i < resources.length; i++) {
+                logger.info(resources[i].getFilename());
+                resourceList.add(resources[i]);
+            }
+        } catch (IOException ex) {
+            ex.printStackTrace();
```

I think it's better to wrap and throw this as a RuntimeException, never really good to swallow errors